flatflow/scheduler/internal/algorithm/regression_test.cc

```diff
@@ -20,95 +20,122 @@
+      sum += coef[i] * std::pow(x, i);
     }
+    return sum;
   }
 
-  double generateDummy(double x, std::vector<double> coef) {
-    double sum = 0;
-    for (std::size_t coefIdx = 0; coefIdx < coef.size(); ++coefIdx) {
-      sum += coef[coefIdx] * std::pow(x, coefIdx);
+  double generate_quadratic(const std::vector<double> &x,
+                            const std::vector<double> &coef) {
+    double sum = coef[0];
+    for (std::size_t i = 1; i < coef.size(); i++) {
+      double temp = 0.0;
+      for (const auto &xi : x) {
+        temp += std::pow(xi, i);
+      }
+      sum += coef[i] * temp;
     }
     return sum;
   }
 
-  std::vector<double> generateCoefficients(std::size_t degree) {
+  std::vector<double> generate_coefficients(std::size_t degree) {
     std::vector<double> coeffs;
-    std::random_device rd;
-    std::mt19937 gen(rd());
-    std::uniform_real_distribution<double> dis(300.0, 400.0);
+    std::random_device random_device;
+    std::mt19937 random_generator(random_device());
+    std::uniform_real_distribution<double> uniform_dist(kMinCoef, kMaxCoef);
     for (std::size_t coefIdx = 0; coefIdx <= degree; ++coefIdx) {
-      coeffs.push_back(dis(gen));
+      coeffs.push_back(uniform_dist(random_generator));
     }
     return coeffs;
   }
 
-  void checkThreshold(const std::vector<double> &modelCofficients,
-                      const std::vector<double> &labelCofficients) {
-    const auto idx = modelCofficients.size() - 1;
-    EXPECT_TRUE(std::abs(modelCofficients[idx] - labelCofficients[idx]) /
-                    labelCofficients[idx] <
-                kThreshold);
+  std::vector<double> random_partition(int x, int node) {
+    if (node == 0) {
+      return {};
+    }
+    std::vector<double> result;
+    result.reserve(node);
+    const auto average = static_cast<double>(x) / node;
+    const auto remainder = x % node;
+    for (std::size_t index = 0; index < node; ++index) {
+      result.push_back(average);
+    }
+    result.back() += remainder;
+    return result;
   }
 
-  static constexpr auto kDatasetSize = static_cast<std::size_t>((1 << 10));
-  static constexpr auto kEpsilon = 1e-4;
-  static constexpr auto kThreshold = 0.01;
-  std::vector<double> X = std::vector<double>(kDatasetSize, 0);
-  std::vector<double> y = std::vector<double>(kDatasetSize, 0);
+  static constexpr auto kLinear = static_cast<std::size_t>(1);
+  static constexpr auto kQuadratic = static_cast<std::size_t>(2);
+  static constexpr auto kEpsilon = static_cast<double>(0.0001);
+  static constexpr auto kThreshold = static_cast<double>(0.01);
+  static constexpr auto kNode = static_cast<int>(4);
+  static constexpr auto kMinRange = static_cast<int>(100);
+  static constexpr auto kMaxRange = static_cast<int>(200);
+  static constexpr auto kMinCoef = static_cast<int>(300);
+  static constexpr auto kMaxCoef = static_cast<int>(400);
+  static constexpr std::size_t kDatasetSize = 100;
```

Comment:
For consistency, define ``kDatasetSize`` with ``static_cast``:
i.e., ``static constexpr auto kDatasetSize = static_cast<std::size_t>(100);``